httpserver/http_connection.cpp

```diff
@@ -498,6 +522,31 @@ void connection_t::process_retrieve(const json& params) {
     const auto pubKey = params["pubKey"].get<std::string>();
     const auto last_hash = params["lastHash"].get<std::string>();
 
+    if (!service_node_.is_pubkey_for_us(pubKey)) {
```

Comment:
There's probably a way to handle "wrong swarm" in a function for both retrieve and store?